### PR TITLE
Added Lint and JSCS to the repo

### PR DESCRIPTION
**Pull request by anmoljagetia**
The core of definitions for 
JSCS - enforces coding guidelines (the definitions are in .jscsrc file)
JSHint - lints JS files, definition is taken from http://jshint.com/docs/ and stored in .jshintrc file.

The rules will have to be adjusted, and then accordingly all the existing files will have to run through the linting process. Works well on Linux and Mac. Might need a few other dependencies on Windows (didn't test on Windows!).

### CODE REVIEW

**Comment by GordonSmith** [comment]
Looks like you didn't add "gulp-notify" to the npm package list (see the travis-ci error above).  While you are in there, can you one (or both) of the lint calls to "npm test" then travis-ci will also run those?

**Comment by anmoljagetia** [comment]
Actually the gulp-notify dependency created a notification queue of about 200 notifications, and seemed like a really bad idea once the tests started to run. I have removed the dependency. I am adding it to the scripts section, in the package.json file. Let's see if the tests pass now.

File: package.json [hunk 1]
```diff
@@ -20,12 +20,19 @@
...
+    "jshint-stylish": "~1.0.2",
+    "gulp-jshint": "~1.10.0",
+    "gulp-jscs": "~1.6.0"
   },
   "repository": {
     "type": "git",
     "url": "https://github.com/hpcc-systems/Visualization"
   },
+  "scripts": {
+    "start": "node server.js",
```
Comment (GordonSmith) [inline package.json]: What is "node server.js" used for?

File: gulpfile.js [hunk 2]
```diff
@@ -82,9 +84,23 @@ gulp.task('build-css', css.bind(null, false));
...
 gulp.task('optimize-css', css.bind(null, true));
 
+gulp.task('jscs', function() {
+    process.stdout.write("JSCS the files...." + '\n');
+    return gulp.src(cfg.src + '/**/*.js')
+        .pipe(jscs());
+});
+
+gulp.task('lint', function() {
+    gulp.src(cfg.src + '/**/*.js')
```
Comment (GordonSmith) [inline gulpfile.js]: Should return the stream for proper async handling.

File: gulpfile.js [hunk 3]
```diff
@@ -82,9 +84,23 @@ gulp.task('build-css', css.bind(null, false));
 
 gulp.task('optimize-css', css.bind(null, true));
 
+gulp.task('jscs', function() {
+    process.stdout.write("JSCS the files...." + '\n');
```
Comment (GordonSmith) [inline gulpfile.js]: Should use "gutil.log" instead of "process.stdout.write" (I think - I have not seen process.stdout.write used in this context before)

**Comment by GordonSmith** [comment]
Closing as I combined it into:  https://github.com/hpcc-systems/Visualization/pull/379

**Pull request closed without merging**